pydra/engine/helpers_file.py

```diff
@@ -71,21 +71,21 @@ def hash_file(afile, chunk_len=8192, crypto=sha256, raise_notfound=True):
     """Compute hash of a file using 'crypto' module."""
     from .specs import LazyField
 
-    if afile is None or isinstance(afile, LazyField) or isinstance(afile, list):
-        return None
-    if not Path(afile).is_file():
-        if raise_notfound:
-            raise RuntimeError('File "%s" not found.' % afile)
-        return None
-
+    try:
+        fp = open(afile, "rb")
+    except FileNotFoundError:
+        if not Path(afile).is_symlink() and raise_notfound:
```

Comment:
and what if `afile` is not a symlink? it also should raise an error

Reply:
Fixed both hash_file and hash_dir in new commit!

Reply:
Sorry for taking so long....